_minutes/2024-12-05-wecg.md

```diff
@@ -0,0 +1,178 @@
+ * [timothy] The fact that we have so many CSP issues might be a good sign of needing content-script-specific CSP in content scripts.
+ * [rob] Not exactly the same issue, in Firefox, we explicitly made the CSP that applies to content scripts the same to the page's for mv3.  If we want to have a different behavior, we should specify exactly the desired behavior.
+ * [giorgio] I agree that the behavior should be specified.
+ * [rob] At a high level, what kind of exceptions do we want to make? We don't make (many) changes to web platform APIs in content scripts. Historically XHR/fetch had different behaviors in MV2. Google folks, are there things beyond CSP that are excepted?
+ * [giorgio] Not a difference between Chrome and Firefox, really. Synchronous XHR limitations are applied to content scripts. Can be worked around using DNR rules. Hack on a hack, but we don't have a meaningful API to address this.
+ * [oliver] From an ergonomics point of view this sounds useful.
+ * [rob] I know Chrome made an effort to have content scripts use the page's CSP, but not well documented. If this is only a CSP thing, then it may be easier to resolve than if there are many other undocumented exceptions.
+ * [oliver] I expect we're unlikely to break this unless there's a significant issue.
+ * [giorgio] If you could document intent to synchronize behavior, I'll stay on the current Firefox behavior (MV2) for now.
+ * [oliver] I'll follow up and leave a comment.
+ * [rob] Apple folks - what does Safari do here?
+ * [timothy] I'll have to test to see what our behavior is in this situation.
+
+[Issue 731](https://github.com/w3c/webextensions/issues/731): Incorrect application of "domainType: thirdParty" in DNR - blocking direct access to iana.org when opening a new tab
+
+ * [dmitry] Title may be inaccurate. We thought the problem was with initiator domain. Five hours ago I found another case where our requests are not blocked. Main issue is with  DNR rules blocking third party requests. We expect that opening a document in a new tab would be a first party request. Seems like a mistake in logic.
+ * [timothy] Agree that it sounds like a bug. Not sure what we do, but I know new page loads are treated as first party.
+ * [kiara] I just tested, Safari doesn't block with these rules.
+ * [rob] Was this observed in Firefox?
+ * [simeon] Yes, noted in the issue description.
+ * [oliver] I know we had some discussion about links from 3rd party applications. Might be worth linking. I know the web's concept of first party directly matches DNR's definition or any other concept.
+ * [rob] I deferred implementation of domainType in Firefox because of the lack of a clear definition of “first-party”. Ended up mirroring Chrome's implementation, which is modeled after adblocker rules, as described at https://bugzilla.mozilla.org/show_bug.cgi?id=1797408#c6.
```

Comment:
```suggestion
 * [rob] I deferred implementation of `domainType` in Firefox because of the lack of a clear definition of "first-party". Ended up mirroring Chrome's implementation, which is modeled after adblocker rules, as described at https://bugzilla.mozilla.org/show_bug.cgi?id=1797408#c6.
```